piecestore/piecestore.go

```diff
@@ -0,0 +1,97 @@
+package piecestore
+
+import (
+	"fmt"
+
+	"github.com/filecoin-project/go-statestore"
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/namespace"
+)
+
+var DSPrefix = "/storagemarket/pieces"
+
+func NewPieceStore(ds datastore.Batching) PieceStore {
+	return &pieceStore{
+		store: statestore.New(namespace.Wrap(ds, datastore.NewKey(DSPrefix))),
+	}
+}
+
+type pieceStore struct {
+	store *statestore.StateStore
+}
+
+func (ps *pieceStore) AddDealForPiece(pieceCID []byte, dealInfo DealInfo) error {
+	// Do we need to de-dupe or anything here?
```

Comment:
This comment is wondering about whether adding duplicate `DealInfo` (or `BlockInfo` below) is necessarily bad or not.  Since it's a local store maybe it's not a big deal.